kobo/apps/subsequences/management/commands/migrate_submission_supplements.py

```
@@ -0,0 +1,283 @@
+                            action=action_id,
+                            params=params,
+                        )
+                        qaf_created += 1
+                    except Exception as e:
+                        self.stderr.write(f'    ERROR: {e}')
+                        qaf_errors += 1
+
+        if not dry_run:
+            self.stdout.write(f'QAFs created: {qaf_created}, errors: {qaf_errors}')
+
+    def _build_params(self, asset: 'Asset', xpath: str, action_id: str) -> list | dict:
```

Ruff (0.15.9)

[error] Undefined name `Asset`

(F821)